File: 9. Check Palindrome - GFG/9.-check-palindrome.py
#User function Template for python3

class Solution:
    def isPalin(self,N):
        #code here
        n = str(N)
        
        if len(n) <= 1:
            return 1
        else:
            if n[0] == n[-1]:
                return self.isPalin(n[1:-1])
            else:
                return 0

#{ 
#  Driver Code Starts
#Initial Template for Python 3

#contributed by RavinderSinghPB
if __name__ == '__main__':
    tcs=int(input())
    
    for _ in range(tcs):
        n=int(input())
        obj = Solution()
        print(int(obj.isPalin(n)))
# } Driver Code Ends

File: Recursion/9. Check Palindrome - GFG/9.-check-palindrome.cpp
// { Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


 // } Driver Code Ends
//User function Template for C++

class Solution{
  public:
    int rev = 0, rem;
    int reverse(int N){
        if(N){
        rem = N % 10;
        rev = rev*10 + rem;
        reverse(N/10);
        }
        return rev;
    }
  
    bool isPalin(int N)
    {
        //Your code here
        //You may use a helper function if you like
        if (N<10){
            return 1;
        }
        return N == reverse(N);
    }
};

// { Driver Code Starts.

int main() {
	int T;
	cin>>T;
	while(T--) 
	{
	    int N;
	    cin>>N;
	    Solution obj;
	    cout<<obj.isPalin(N)<<endl;
	}
	return 0;
}  // } Driver Code Ends